### PR TITLE
Log which benchmark failed at the point the error is reported.

### PR DESCRIPTION
**Pull request by vext01**
This makes scanning the logs much, much easier for me.

### CODE REVIEW

File: krun/util.py [hunk 1]
```diff
@@ -248,7 +248,7 @@ def check_and_parse_execution_results(stdout, stderr, rc, config,
     if json_exn or rc != 0:
         # Something went wrong
         rule = 50 * "-"
-        err_s = ("Benchmark returned non-zero or emitted invalid JSON.\n")
+        err_s = ("Benchmark '%s' returned non-zero or emitted invalid JSON.\n" % key)
```
Comment (vext01) [inline krun/util.py:251]: This hunk is the crux of this PR. The rest is updating call-sites.

File: krun/scheduler.py [hunk 2]
```diff
@@ -383,7 +383,7 @@ def run(self, mailer, dry_run=False):
             try:
                 self.sched.platform.check_throttle_counts(self.sched.manifest)
                 measurements = util.check_and_parse_execution_results(
-                    stdout, stderr, rc, self.sched.config, instrument=vm_def.instrument)
+                    stdout, stderr, rc, self.sched.config, self.key, instrument=vm_def.instrument)
```
Comment (ltratt) [inline krun/scheduler.py:386]: I might be missing out on something, but this is only the assignment of `key` to a field -- nothing seems to read from it?

Comment (vext01) [reply krun/scheduler.py:386]: It's passing `self.key` as a function argument so that we can print it in the log.

Comment (ltratt) [reply krun/scheduler.py:386]: I'm an idiot.

Comment (vext01) [reply krun/scheduler.py:386]: Don't sweat it. Functions with that many args become hard to scan :P

**Pull request merged**